scripts/py_matter_idl/matter_idl/generators/java/ChipClusters_java.jinja

```diff
@@ -126,22 +179,68 @@ public class ChipClusters {
      * empty optional, the default timeout will be used.
      */
     public void setCommandTimeout(Optional<Long> timeoutMillis) {
-      setCommandTimeout(chipClusterPtr, timeoutMillis);
+      this.timeoutMillis = timeoutMillis;
     }
 
-    private native void setCommandTimeout(long clusterPtr, Optional<Long> timeoutMillis);
-
     /** Returns the current timeout (in milliseconds) for commands sent through this cluster. */
     public Optional<Long> getCommandTimeout() {
-      Optional<Long> timeout = getCommandTimeout(chipClusterPtr);
-      return timeout == null ? Optional.empty() : timeout;
+      return timeoutMillis == null ? Optional.empty() : timeoutMillis;
     }
 
-    private native Optional<Long> getCommandTimeout(long clusterPtr);
-
+    @Deprecated
     public abstract long initWithDevice(long devicePtr, int endpointId);
 
-    public native void deleteCluster(long chipClusterPtr);
+    protected void readAttribute(
```

Comment:
What is the difference of the readAttribute in BaseChipCluster and readAttribute in deviceController?

Reply:
As of now, it is not possible to directly call methods of the CHIPDeviceController class. There is no object of CHIPDeviceController class in ChipClusters class. This part is currently under consideration.

Reply:
I agree